### ISSUE TITLE
Add support for AWS Cloudfront

### ISSUE THREAD

**Issue by alikhajeh1** [infracost/infracost#230]
Pricing details: https://aws.amazon.com/cloudfront/pricing/
Terraform resources: aws_cloudfront_*

**Comment by sinabakh** [comment]
@alikhajeh1 From the terraform's configs point of view, there is nothing we can calculate. Do we want to support it and show no cost for now? If that's the case I can work on it.

**Comment by alikhajeh1** [comment]
@sinabakh we now consider https://www.infracost.io/docs/usage_based_resources out of beta, so we can add [level 1 and 2](https://github.com/infracost/infracost/blob/master/CONTRIBUTING.md) support for it. The location-based pricing is challenging as it'll mean we show lots of prices, our options are:
1. Show all prices unless the users is using the provider, in which case we show the relevant prices based on the locations they enter.
2. Show US prices unless the users is using the provider, in which case we show the relevant prices based on the locations they enter.

I think #1 works better for cloudfront since CDNs are for users to distribute the content worldwide so them knowing the world-wide prices seems more useful? What do you think?

**Comment by sinabakh** [comment]
@alikhajeh1 Yes I agree, it seems a little complicated to handle all these regions. I will work on it if you don't mind.
The only challenge I can think of is the key names for usages in regions like `Hong Kong, Philippines, Singapore, South Korea, Taiwan, & Thailand`.

**Comment by sinabakh** [comment]
This issue was stale for a while since it's so complicated and I was postponing it for some time.
However, regarding the regions in pricing (take *Regional Data Transfer Out to Origin (per GB)* for instance). Here are the grouped regions:

  | United States, Mexico, & Canada | Europe & Israel | South Africa, Kenya, & Middle East | South America | Japan | Australia & New Zealand | Hong Kong, Philippines, Singapore, South Korea, Taiwan, & Thailand | India
-- | -- | -- | -- | -- | -- | -- | -- | --
All Data Transfer | $0.020 | $0.020 | $0.060 | $0.125 | $0.060 | $0.080 | $0.060 | $0.160

But using the pricing query, we got different regions:
```js
query {
  products(
    filter: {
      vendorName: "aws",
      service: "AmazonCloudFront",
      attributeFilters: [
        { key: "transferType", value: "CloudFront to Origin" },
      ]
    },
  ) {
    attributes { key, value }
    prices { USD, priceHash }
  }
}
```

Here is the list of unique regions:
```
Asia Pacific
Japan
South America
South Africa
Europe
Australia
Middle East
United States
India
Canada
```

As you can see there are differences. How should we list the costs for users? I believe it should be compatible with Amazon pricing page but they could possibly change it anytime and we might not get notified. Also, how can we name the keys for our own provider for these regions?

@alikhajeh1 @aliscott  I will be more than happy if you could give your opinions.

**Comment by aliscott** [comment]
I agree with you that we should show the prices in the same way they are on the AWS pricing page. The groupings are probably less likely to change, so I'm not too worried about that at the moment. Here's how I would do the mappings:

```
United States: United States, Mexico, & Canada
Europe:        Europe & Israel
South Africa:  South Africa, Kenya, & Middle East
South America: South America
Japan:         Japan
Australia:     Australia & New Zealand
Asia Pacific:  Hong Kong, Philippines, Singapore, South Korea, Taiwan, & Thailand
India:         India
```

For the usage data I'd start with something like:

```
united_states_data_transfer_gb
europe_data_transfer_gb
south_africa_data_transfer_gb
south_america_data_transfer_gb
japan_data_transfer_gb
australia_data_transfer_gb
asia_pacific_data_transfer)db
india_data_transfer_gb
```

I'd also consider adding `middle_east_data_transfer_gb` and summing it with `south_africa_data_transfer_gb`, what do you think? We should try and keep it as simple as we can.

**Comment by sinabakh** [comment]
@aliscott Looks good to me, the usage data might be a little odd for the user but that's the best we can do.

**Comment by alikhajeh1** [comment]
@syphernl just FYI that support for aws_ssm_parameter and aws_ssm_activation was just released in v0.7.15 (I noticed you +1'd the `aws_cloudfront_distribution` resource too, that was released in the last version but I forgot to move [the ticket](https://github.com/infracost/infracost/issues/230) to released).
